pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/operands/FilterOperand.java

```diff
@@ -95,6 +106,22 @@ public Boolean apply(Object[] row) {
   @Override
   public abstract Boolean apply(Object[] row);
 
+  private static class BooleanInputRef extends FilterOperand {
+    private final RexExpression.InputRef _inputRef;
+
+    public BooleanInputRef(RexExpression.InputRef inputRef, DataSchema dataSchema) {
+      Preconditions.checkState(dataSchema.getColumnDataType(inputRef.getIndex())
+          == DataSchema.ColumnDataType.BOOLEAN);
+      _inputRef = inputRef;
+    }
+
+    @Override
+    public Boolean apply(Object[] row) {
+      // DataSchema.ColumnDataType.BOOLEAN.getStoredType() == INT
+      return (int) row[_inputRef.getIndex()] != 0;
```

Comment:
This seems less intuitive. If we are returning Boolean, why do we need to typecast ?

Reply:
b/c the underlying row object in pinot is stored as integer
- directly return `Object` as a boolean will do a null check
- but I was also thinking it could've been an actual boolean object, in which case this will work but not efficient